opennms-webapp-rest/src/main/java/org/opennms/web/rest/v2/NodeLinkRestService.java

```diff
@@ -0,0 +1,249 @@
+
+    @Autowired
+    private EnLinkdElementFactoryInterface enLinkdElementFactory;
+
+    @Autowired
+    private NodeDao m_nodeDao;
+
+    @Override
+    public EnlinkdDTO getEnlinkd(String nodeId) {
+        return new EnlinkdDTO()
+                .withLldpLinkNodeDTOs(getLldpLinks(nodeId))
+                .withBridgeLinkNodeDTOS(getBridgelinks(nodeId))
+                .withCdpLinkNodeDTOS(getCdpLinks(nodeId))
+                .withOspfLinkNodeDTOS(getOspfLinks(nodeId))
+                .withIsisLinkNodeDTOS(getIsisLinks(nodeId))
+                .withLldpElementNodeDTO(getLldpelem(nodeId))
+                .withCdpElementNodeDTO(getCdpelem(nodeId))
+                .withOspfElementNodeDTO(getOspfelem(nodeId))
+                .withIsisElementNodeDTO(getIsiselem(nodeId))
+                ;
+    }
+
+    @Override
+    public List<LldpLinkNodeDTO> getLldpLinks(String nodeId) {
+        return enLinkdElementFactory.getLldpLinks(getNodeIdInDB(nodeId)).stream().map(this::mapLldpLindNodeToDTO).collect(Collectors.toList());
+    }
+
+    @Override
+    public List<BridgeLinkNodeDTO> getBridgelinks(String nodeId) {
+        return enLinkdElementFactory.getBridgeLinks(getNodeIdInDB(nodeId)).stream().map(this::mapBridgeLinkNodeToDTO).collect(Collectors.toList());
+    }
+
+    @Override
+    public List<CdpLinkNodeDTO> getCdpLinks(String nodeId) {
+        return enLinkdElementFactory.getCdpLinks(getNodeIdInDB(nodeId)).stream().map(this::mapCdpLinkNodeToDTO).collect(Collectors.toList());
+    }
+
+    @Override
+    public List<OspfLinkNodeDTO> getOspfLinks(String nodeId) {
+        return enLinkdElementFactory.getOspfLinks(getNodeIdInDB(nodeId)).stream().map(this::mapOspfLinkNodeToDTO).collect(Collectors.toList());
+    }
+
+    @Override
+    public List<IsisLinkNodeDTO> getIsisLinks(String nodeId) {
+        return enLinkdElementFactory.getIsisLinks(getNodeIdInDB(nodeId)).stream().map(this::mapIsisLinkNodeToDTO).collect(Collectors.toList());
+    }
+
+    @Override
+    public LldpElementNodeDTO getLldpelem(String nodeId) {
+        return mapLldElementNodeToDTO(enLinkdElementFactory.getLldpElement(getNodeIdInDB(nodeId)));
+    }
+
+    @Override
+    public CdpElementNodeDTO getCdpelem(String nodeId) {
+        return mapCdpElementNodeToDTO(enLinkdElementFactory.getCdpElement(getNodeIdInDB(nodeId)));
+    }
+
+    @Override
+    public OspfElementNodeDTO getOspfelem(String nodeId) {
+        return mapOspfElementNodeToDTO(enLinkdElementFactory.getOspfElement(getNodeIdInDB(nodeId)));
+    }
+
+    @Override
+    public IsisElementNodeDTO getIsiselem(String nodeId) {
+        return mapIsisElementNodeToDTO(enLinkdElementFactory.getIsisElement(getNodeIdInDB(nodeId)));
+    }
+
+    private int getNodeIdInDB(String nodeId) {
+        OnmsNode onmsNode =  m_nodeDao.get(nodeId);
+        if (onmsNode == null) {
+            throw new NoSuchElementException("Node Id does not exist in database: " + nodeId);
```

Comment:
`NoSuchElementException("Not able to find node with criteria :  " + nodeCriteria)`